catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/action/provider/RegistryPublicationActionProvider.java

```diff
@@ -0,0 +1,261 @@
+
+    private static final String PUBLISH_TITLE = "Publish to ";
+
+    private static final String UNPUBLISH_DESCRIPTION = "Unpublishes this nodes information from ";
+
+    private static final String UNPUBLISH_OPERATION = "unpublish";
+
+    private static final String UNPUBLISH_TITLE = "Unpublish from ";
+
+    private static final String HTTP_POST = "HTTP_POST";
+
+    private static final String HTTP_DELETE = "HTTP_DELETE";
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private static final String CREATED_TOPIC = "ddf/catalog/event/CREATED";
+
+    private static final String UPDATED_TOPIC = "ddf/catalog/event/UPDATED";
+
+    private static final String DELETED_TOPIC = "ddf/catalog/event/DELETED";
+
+    private String providerId;
+
+    private ConfigurationAdmin configAdmin;
+
+    private FederationAdminService federationAdminService;
+
+    private List<RegistryStore> registryStores;
+
+    private Map<String, List<String>> publications = new HashMap<>();
+
+    @Override
+    public <T> List<Action> getActions(T subject) {
+        String registryId = getRegistryId(subject);
+        if (registryId == null) {
+            return Collections.emptyList();
+        }
+        List<Action> actions = new ArrayList<>();
+        List<String> currentPublications = publications.get(registryId);
+
+        for (RegistryStore registry : registryStores) {
+            //can't publish to yourself
+            if (!registry.isPushAllowed() || StringUtils.isBlank(registry.getRegistryId()) || registryId.equals(registry.getRegistryId())) {
+                continue;
+            }
+
+            if (currentPublications.contains(registry.getId())) {
+                actions.add(getAction(registryId, registry.getId(), false));
+            } else {
+                actions.add(getAction(registryId, registry.getId(), true));
+            }
+        }
+
+        return actions;
+    }
+
+    @Override
+    public String getId() {
+        return providerId;
+    }
+
+    @Override
+    public <T> boolean canHandle(T subject) {
+        return getRegistryId(subject) != null;
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null || !mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            return;
+        }
+
+        String registryId = mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+        Attribute locationAttribute =
+                mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        List<String> locations = new ArrayList<>();
+        if (locationAttribute != null) {
+            locations = locationAttribute.getValues()
+                    .stream()
+                    .map(Object::toString)
+                    .collect(Collectors.toList());
+        }
```

Comment:
Do we need to consider thread safety when putitng and removing from `publications`?